src/librustdoc/html/static/js/search.js

```diff
@@ -1,6 +1,55 @@
 /* global addClass, getNakedUrl, getSettingValue, hasOwnPropertyRustdoc, initSearch, onEach */
 /* global onEachLazy, removeClass, searchState, updateLocalStorage */
 
+/* eslint-disable */
+
+/**
+ * @typedef {{
+ *   doc: string,
```

Comment:
I'm not sure how useful this comment is...

Reply:
The main goal for this comment is to describe the schema of these input rows from search-index.js, in a way that can be automatically checked by Closure Compiler.

I went ahead and wrote a longer comment trying to describe the schema better. I had to put "is a mystery" for some of the fields because I couldn't figure them out. Maybe you can help me fill in the details. :-)